lightningd/lightningd.c

```diff
@@ -1115,7 +1115,7 @@ int main(int argc, char *argv[])
 	/*~ Pull peers, channels and HTLCs from db. Needs to happen after the
 	 *  topology is initialized since some decisions rely on being able to
 	 *  know the blockheight. */
-	unconnected_htlcs_in = load_channels_from_wallet(ld);
+	unconnected_htlcs_in = notleak(load_channels_from_wallet(ld));
```

Comment:
nit: on the usage of the `notleak` function, but I agree that this is not a real or maybe always a leak!

I propose the alternative to this in https://github.com/ElementsProject/lightning/pull/5933

The main reason that I would like to avoid using the notleak function is that the report is right, in the `load_channels_from_wallet` we leave the map unfree while we are aborting (this cause breaks our normal program workflow). Also we almost never initialize a variable when we declare it as we do in https://github.com/ElementsProject/lightning/pull/5933/files#diff-1656b5a11d89a9b988706ec009c8e240cacb94009fc74069757845060b2bfeb8L2223